### PR TITLE
fix some potential memleaks and null pointer dereferences

### PR DESCRIPTION
**Pull request by quink-black**

### CODE REVIEW

**Comment by quink-black** [comment]
Ping for review, thanks!

File: cocos/scripting/js-bindings/manual/jsb_global.cpp [hunk 1]
```diff
@@ -831,36 +831,42 @@ bool jsb_global_load_image(const std::string& path, const se::Value& callbackVal
     std::shared_ptr<se::Value> callbackPtr = std::make_shared<se::Value>(callbackVal);
 
     auto initImageFunc = [path, callbackPtr](const std::string& fullPath, unsigned char* imageData, int imageBytes, const std::string& errorMsg){
-        Image* img = new (std::nothrow) Image();
+        std::shared_ptr<uint8_t> imageDataGuard(imageData, free);
```
Comment (PatriceJiang) [inline cocos/scripting/js-bindings/manual/jsb_global.cpp:834]: I suggest using `unique_ptr`

File: cocos/scripting/js-bindings/manual/jsb_global.cpp [hunk 2]
```diff
@@ -831,36 +831,42 @@ bool jsb_global_load_image(const std::string& path, const se::Value& callbackVal
...
+                loadSucceed = img->initWithImageData(imageDataGuard.get(), imageBytes);
+                imageDataGuard = nullptr;
             }
             else
             {
                 loadSucceed = img->initWithImageFile(fullPath);
             }
 
-            struct ImageInfo* imgInfo = nullptr;
+            std::shared_ptr<ImageInfo> imgInfo;
```
Comment (PatriceJiang) [inline cocos/scripting/js-bindings/manual/jsb_global.cpp:863]: use `unique_ptr`

File: cocos/scripting/js-bindings/manual/jsb_global.cpp (hunk 1 above)
Comment (quink-black) [reply cocos/scripting/js-bindings/manual/jsb_global.cpp:834]: It's not easy to capture unique_ptr with C++11, ref.

https://stackoverflow.com/questions/8236521/how-to-capture-a-unique-ptr-into-a-lambda-expression

We can do that after cocos update to C++14.

**Comment by quink-black** [comment]
可以merge了吗？

**Pull request merged**